In [80]:
import requests

In [81]:
response = requests.get("https://news.naver.com/section/105")

In [82]:
data = response.text.split('data-template-id="SECTION_HEADLINE">')[1]
data

'\n\t<div class="sa_head">\n\t\t<span class="sa_head_inner">\n\t\t\t<a href="#" class="sa_head_link _TOGGLE" data-classValue="is_hidden" data-target="_SECTION_HEADLINE_INFO_hhuxi" data-clk="guide">헤드라인 뉴스 <i class="sa_head_icon">안내</i></a>\n\t\t</span>\n\t\t<div id="_SECTION_HEADLINE_INFO_hhuxi" class="sa_head_layer is_hidden">\n\t\t\t<p class="sa_head_layer_p">각 헤드라인의 기사와 배열 순서는 개인화를 반영해 추천되며, 기사묶음의 대표 기사는 구독 언론사 중심으로 제공됩니다. 기사 수량이 표기된 기사 우측 하단의 파란색 아이콘을 클릭하면 기사묶음을 확인할 수 있고 기사묶음과 기사묶음 타이틀도 기사 내용을 기반으로 자동 추출됩니다.</p>\n\t\t\t<a href="https://media.naver.com/algorithm" class="sa_head_layer_go">알고리즘 자세히 보기</a>\n\t\t\t<div class="sa_head_layer_close">\n\t\t\t\t<button type="button" class="_TOGGLE" data-classValue="is_hidden" data-target="_SECTION_HEADLINE_INFO_hhuxi">닫기</button>\n\t\t\t</div>\n\t\t</div>\n\t</div>\n\t<ul id="_SECTION_HEADLINE_LIST_hhuxi" class="sa_list">\n\t\t<li class="sa_item _SECTION_HEADLINE">\n\t\t\t<div class="sa_item_inner">\n\t\t\t\t<div class="sa_item_flex _LAZY_LO

In [83]:
data = data.split('sa_list">')[1]
data = data.split(r'</ul>')[0]
data

'\n\t\t<li class="sa_item _SECTION_HEADLINE">\n\t\t\t<div class="sa_item_inner">\n\t\t\t\t<div class="sa_item_flex _LAZY_LOADING_WRAP">\n\t\t\t\t\t<div class="sa_thumb _LAZY_LOADING_ERROR_HIDE">\n\t\t\t\t\t\t<div class="sa_thumb_inner">\n\t\t\t\t\t\t\t<a href="https://n.news.naver.com/mnews/article/028/0002749282" class="sa_thumb_link _NLOG_IMPRESSION" data-clk="itn.clart" data-imp-gdid="88000103_000000000000000002749282" data-imp-url="https://n.news.naver.com/mnews/article/028/0002749282" data-imp-index="1">\n\t\t\t\t\t\t\t\t<img data-src="https://mimgnews.pstatic.net/image/origin/028/2025/06/04/2749282.jpg?type&#x3D;nf220_150" class="_LAZY_LOADING _LAZY_LOADING_INIT_HIDE" width="110" height="75" alt="삼성전자, 다음달 갤럭시Z 폴드7 출시…“울트라급 될 것”" style="display: none;">\n\t\t\t\t\t\t\t</a>\n\t\t\t\t\t\t</div>\n\t\t\t\t\t</div>\n\t\t\t\t\t<div class="sa_text">\n\t\t\t\t\t\t<a href="https://n.news.naver.com/mnews/article/028/0002749282" class="sa_text_title _NLOG_IMPRESSION" data-clk="itn.clart" da

<img class=\"_LAZY_LOADING _LAZY_LOADING_INIT_HIDE\" width=\"110\" height=\"75\" alt=\"카이스트 사족보행 로봇 '라이보', 고속 질주에 벽까지 달린다\" style=\"\" src=\"https://mimgnews.pstatic.net/image/origin/421/2025/06/03/8291585.jpg?type=nf220_150\">

In [84]:
import re
# p = re.compile(r"<img.+?src=['\"][^'\"]+['\"][^>]*>", re.I|re.S|re.X)
p = re.compile(r"<img.+?src=['\"]([^'\"]+)['\"][^>]*>", re.I|re.S|re.X)
m = p.search(data)
m

<re.Match object; span=(489, 726), match='<img data-src="https://mimgnews.pstatic.net/image>

In [85]:
images = p.findall(data)
images

['https://mimgnews.pstatic.net/image/origin/028/2025/06/04/2749282.jpg?type&#x3D;nf220_150',
 'https://mimgnews.pstatic.net/image/origin/009/2025/06/04/5503343.jpg?type&#x3D;nf220_150',
 'https://mimgnews.pstatic.net/image/origin/020/2025/06/04/3639123.jpg?type&#x3D;nf220_150',
 'https://mimgnews.pstatic.net/image/origin/003/2025/06/04/13284628.jpg?type&#x3D;nf220_150',
 'https://mimgnews.pstatic.net/image/origin/052/2025/06/04/2201579.jpg?type&#x3D;nf220_150',
 'https://mimgnews.pstatic.net/image/origin/016/2025/06/04/2480286.jpg?type&#x3D;nf220_150',
 'https://mimgnews.pstatic.net/image/origin/009/2025/06/04/5503271.jpg?type&#x3D;nf220_150',
 'https://mimgnews.pstatic.net/image/origin/001/2025/06/04/15430526.jpg?type&#x3D;nf220_150']

In [86]:
iter = p.finditer(data)
iter

In [87]:
# next(iter)
for m in iter:
    print(m)

<re.Match object; span=(489, 726), match='<img data-src="https://mimgnews.pstatic.net/image>
<re.Match object; span=(2519, 2768), match='<img data-src="https://mimgnews.pstatic.net/image>
<re.Match object; span=(4575, 4817), match='<img data-src="https://mimgnews.pstatic.net/image>
<re.Match object; span=(6616, 6850), match='<img data-src="https://mimgnews.pstatic.net/image>
<re.Match object; span=(8640, 8890), match='<img data-src="https://mimgnews.pstatic.net/image>
<re.Match object; span=(10716, 10956), match='<img data-src="https://mimgnews.pstatic.net/image>
<re.Match object; span=(12763, 13007), match='<img data-src="https://mimgnews.pstatic.net/image>
<re.Match object; span=(14817, 15061), match='<img data-src="https://mimgnews.pstatic.net/image>


In [88]:
iter = p.finditer(data)
iter

In [89]:
iter = p.finditer(data)
for m in iter:
    # print(f"group:{m.group()} / start:{m.start()} / end:{m.end()} / span:{m.span()}")
    print(f"url:{m.group(1)}")

url:https://mimgnews.pstatic.net/image/origin/028/2025/06/04/2749282.jpg?type&#x3D;nf220_150
url:https://mimgnews.pstatic.net/image/origin/009/2025/06/04/5503343.jpg?type&#x3D;nf220_150
url:https://mimgnews.pstatic.net/image/origin/020/2025/06/04/3639123.jpg?type&#x3D;nf220_150
url:https://mimgnews.pstatic.net/image/origin/003/2025/06/04/13284628.jpg?type&#x3D;nf220_150
url:https://mimgnews.pstatic.net/image/origin/052/2025/06/04/2201579.jpg?type&#x3D;nf220_150
url:https://mimgnews.pstatic.net/image/origin/016/2025/06/04/2480286.jpg?type&#x3D;nf220_150
url:https://mimgnews.pstatic.net/image/origin/009/2025/06/04/5503271.jpg?type&#x3D;nf220_150
url:https://mimgnews.pstatic.net/image/origin/001/2025/06/04/15430526.jpg?type&#x3D;nf220_150
